src/Fleck/Handlers/Hybi13Handler.cs

```diff
@@ -47,7 +48,36 @@ public static byte[] FrameData(byte[] payload, FrameType frameType)
             
             return memoryStream.ToArray();
         }
-        
+
+        public static byte[] FrameData(ArraySegment<byte> payload, FrameType frameType)
+        {
+            var memoryStream = new MemoryStream();

```

Comment:
Another allocation that could be skipped is if the MemoryStream is initialized with a starting buffer then `memoryStream.toArray` will just return the underlying array. To determine the buffer length, we'd just need to defer creating it until after calculating `lengthBytes`.

Reply:
The problem with that is, the returned array from `MemoryStream.ToArray()` still has the length of the length of the underlying buffer (See <https://replit.com/@kuzux/memorystreamarray#main.cs>). The length of that array is then used by the `SendBytes` function. If we want it to function correctly, we'll need to initialize the underlying buffer for each message.

Unless you were talking about the extra allocation that happens after writing a large amount of bytes to the stream.